### PR TITLE
[WIP] Normalize transformed space to [0,1]^p

### PR DESCRIPTION
**Pull request by glouppe**
This is an exploratory branch for #160

### CODE REVIEW

**Comment by glouppe** [comment]
For further reference, Spearmint uses Beta warping by default. See https://github.com/HIPS/Spearmint/blob/master/spearmint/models/gp.py#L319

**Comment by MechCoder** [comment]
Should we keep both changes independent? I was hacking on the beta warp before the package was created, but then I had abandoned the effort. I would like to continue on that right now, now that we have a stable API.

**Comment by MechCoder** [comment]
> Spearmint uses Beta warping by default.

No surprises with regards to that. The authors of Spearmint wrote that paper as well :-)

**Comment by betatim** [comment]
Could be separate PRs. Would you have to wait for this to be merged?

**Comment by MechCoder** [comment]
Not really. The beta warp transforms the space to [0, 1] (since it is a CDF)

**Comment by betatim** [comment]
I only had a quick glance at the paper, it looked like beta warping needed the variables to be in [0, 1] already.

We should provide a simple transform which scales everything to [0, 1] linearly and have beta warping as a separate tool.

**Comment by MechCoder** [comment]
Seems like it. Would you like to continue this PR?

On a side note, do you understand how to normalize to [0, 1], when the bounds are say (10**-5, 10**5)?

**Comment by MechCoder** [comment]
Hmm. I think we just provide the input to kernel after sampling uniformly from [0, 1] and do whatever is necessary in the inverse transform.

**Comment by betatim** [comment]
> Would you like to continue this PR?

Wasn't planning to

**Comment by glouppe** [comment]
I'll resume working on this in the next days or so.

**Comment by MechCoder** [comment]
While you are it, do you mind recording the values that you get while comparing the performance. I feel it will be useful while handling (https://github.com/scikit-optimize/scikit-optimize/issues/205)

File: skopt/space.py [hunk 1]
```diff
@@ -23,6 +23,23 @@ def inverse_transform(self, Xt):
         return Xt
 
 
+class _Normalize:
```
Comment (MechCoder) [inline skopt/space.py]: Should we rename this to maybe `HypercubeTransform` and make it public? `Normalize` can be confused with unit norm normalization.

**Comment by MechCoder** [comment]
@glouppe This seems quite ready. What is left to do?

**Comment by betatim** [comment]
Do you want to complete it @glouppe or can someone else cherry-pick the commits?

**Comment by glouppe** [comment]
You can go ahead and take it, but given the other changes we had meanwhile, it might be simpler to just start from scratch.

**Comment by MechCoder** [comment]
just fyi, working on this now

**Comment by MechCoder** [comment]
see https://github.com/scikit-optimize/scikit-optimize/pull/237

**Pull request closed without merging**